src/lsm/lsm_manager.c

```diff
@@ -658,6 +658,14 @@ __wt_lsm_manager_push_entry(WT_SESSION_IMPL *session,
 	manager = &S2C(session)->lsm_manager;
 
 	/*
+	 * Don't allow any work units unless a tree is active, this avoids
+	 * races on shutdown between clearing out queues and pushing new
+	 * work units.
+	 */
+	if (!F_ISSET(lsm_tree, WT_LSM_TREE_ACTIVE))
```

Comment:
@agorrod I think this narrows the window but does not close it.  The path I'm thinking of is:
T1: In `__wt_lsm_manager_push_entry` and `WT_LSM_TREE_ACTIVE` is set so we pass the new test above.
T1: Gets descheduled, say, during the calloc call at line 685.
T2: Calls `__lsm_tree_close` and clears `WT_LSM_TREE_ACTIVE`.
T2: Calls `__lsm_tree_discard` and passes the new assertion and proceeds to close the tree.
T1: Runs again, bumps the `lsm_tree->queue_ref` and pushes the work unit.  I think at this point it's in the same situation as before.